FaunaDB.Client/Client/DefaultClientIO.cs

```diff
@@ -48,9 +54,14 @@ async Task<RequestResult> DoRequestAsync(HttpMethodKind method, string path, str
 
             var startTime = DateTime.UtcNow;
 
-            var message = new HttpRequestMessage(new HttpMethod(method.Name()), path);
+            var message = new HttpRequestMessage(new HttpMethod(method.Name()), $"{endpoint}{path}");
             message.Content = dataString;
             message.Headers.Authorization = authHeader;
+            message.Headers.AcceptEncoding.Add(new StringWithQualityHeaderValue("gzip"));
+            message.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+            message.Headers.Add("X-FaunaDB-API-Version", "2.7");
+            message.Headers.Add("X-Fauna-Driver", "csharp");
+            message.SetTimeout(timeout);
```

Comment:
why are you setting timeout here? we are already setting this on `HttpClient` creation

Reply:
I removed that, because the user may pass their own. So you need to set it on the HttpRequestMessage for each call. The user may have their own timeout or the default. Don't want to change it and mess up how their other requests deal with them